### PR TITLE
meta: Make sure meta key always less then other key

### PR DESCRIPTION
**Pull request by ngaut**
Now meta key start with 0x0.

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: meta/meta.go [hunk 1]
```diff
@@ -21,19 +21,26 @@ import (
...
-	SchemaMetaVersionKey = []byte("mSchemaVersion")
+	SchemaMetaVersionKey = MakeMetaKey("mSchemaVersion")
+
+	// globalIDPrefix is used as key of global ID
+	globalIDKey = MakeMetaKey("mNextGlobalID")
 )
 
+// MakeMetaKey creates meta key
+func MakeMetaKey(key string) []byte {
+	return append([]byte{0x0}, key...)
```
Comment (shenli) [inline meta/meta.go]: Why force metakey starts with 0x0?

File: meta/meta.go [hunk 2]
```diff
@@ -21,19 +21,26 @@ import (
...
 	// TableMetaPrefix is the prefix for table meta key prefix.
-	TableMetaPrefix = "mTable:"
-)
+	TableMetaPrefix = MakeMetaKey("mTable:")
 
-var (
 	nextGlobalIDPrefix = []byte("mNextGlobalID")
 	// SchemaMetaVersionKey is used as lock for changing schema
-	SchemaMetaVersionKey = []byte("mSchemaVersion")
+	SchemaMetaVersionKey = MakeMetaKey("mSchemaVersion")
```
Comment (qiuyesuifeng) [inline meta/meta.go]: It is better to add an empty line between line 31 and 32.

Comment (qiuyesuifeng) [inline meta/meta.go]: Seems nextGlobalIDPrefix can be removed?

File: meta/meta.go (hunk 1 above)
Comment (ngaut) [inline meta/meta.go]: To separate from meta and data. Also we would never scan meta while scanning data.

**Comment by ngaut** [comment]
PTAL

File: table/tables/tables.go [hunk 3]
```diff
@@ -534,6 +534,10 @@ func (t *Table) IterRecords(ctx context.Context, startKey string, cols []*column
 	}
 	defer it.Close()
 
+	if !it.Valid() {
```
Comment (c4pt0r) [inline table/tables/tables.go]: This check works if and only if the data keys are in the bottom of whole table. But I didn't seem any guarantee about it.

Comment (ngaut) [inline table/tables/tables.go]: It's invalid when meeting empty table.

Comment (c4pt0r) [inline table/tables/tables.go]: But if there'll be only 2 types of key: meta and data, it's OK.

Comment (ngaut) [inline table/tables/tables.go]: Below debug message need iterator to be valid.

**Comment by c4pt0r** [comment]
LGTM

**Pull request merged**